pkg/client/client_test.go

```diff
@@ -1654,6 +1656,36 @@ U5wwSivyi7vmegHKmblOzNVKA5qPO8zWzqBC
 				Expect(err).To(HaveOccurred())
 			})
 
+			It("should update the resource when deleting if it receives a response", func() {
+				cl, err := client.New(cfg, client.Options{})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(cl).NotTo(BeNil())
+
+				By("initially creating a Node")
+				node, err := clientset.CoreV1().Nodes().Create(ctx, node, metav1.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("adding a finalizer we prevent the node from being deleted immediately")
+				controllerutil.AddFinalizer(node, "example.com/test")
+				node, err = clientset.CoreV1().Nodes().Update(ctx, node, metav1.UpdateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				By("deleting the Node")
+				nodeName := node.Name
+				err = cl.Delete(context.TODO(), node)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(node.ObjectMeta.DeletionTimestamp).NotTo(BeNil())
```

Comment:
Can we also test this with `unstructured.Unstructured` and `metav1.MetaObject`, please?

Reply:
Unstructured works... metav1.MetaObject will require a fair bit more work.

https://github.com/kubernetes-sigs/controller-runtime/blob/c5bb1d4d8e7d0e49bb06f8276785e7f7cbcbf2b8/pkg/client/metadata_client.go#L67

Not sure if this is relevant? https://github.com/kubernetes-sigs/controller-runtime/blob/c5bb1d4d8e7d0e49bb06f8276785e7f7cbcbf2b8/pkg/client/metadata_client.go#L30

Reply:
I think we should also implement this for the metadata client. It's already tricky that we change a nuance like this, but I think it's a lot worse if our clients behave inconsistently